source/common/http/http2/codec_impl.cc

```diff
@@ -546,6 +546,14 @@ void ConnectionImpl::StreamImpl::encodeDataHelper(Buffer::Instance& data, bool e
   }
 }
 
+void ConnectionImpl::ServerStreamImpl::resetStream(StreamResetReason reason) {
+  StreamImpl::resetStream(reason);
+  // Clear the downstream on the account since we're resetting the downstream.
+  if (buffer_memory_account_) {
+    buffer_memory_account_->clearDownstream();
+  }
```

Comment:
clear before calling base class method may make sense.  But isn't this covered by the call in ServerStreamImpl::destroy()?

Reply:
Now call clear before the base class method.

`StreamImpl::resetStream`  can sometimes defer the reset (if end stream already but not end stream sent). By having this in the SeverStreamImpl::resetStream, we know the BufferFactory won't try to reset the stream again since clearing downstream unregister's the account.